File: dart/com/spoonacular/client/model/inline_response20047_recommended_wines.dart
part of openapi.api;

class InlineResponse20047RecommendedWines {
  
  int id = null;
  
  String title = null;
  
  num averageRating = null;
  
  String description = null;
  
  String imageUrl = null;
  
  String link = null;
  
  String price = null;
  
  int ratingCount = null;
  
  num score = null;
  InlineResponse20047RecommendedWines();

  @override
  String toString() {
    return 'InlineResponse20047RecommendedWines[id=$id, title=$title, averageRating=$averageRating, description=$description, imageUrl=$imageUrl, link=$link, price=$price, ratingCount=$ratingCount, score=$score, ]';
  }

  InlineResponse20047RecommendedWines.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['id'] == null) {
      id = null;
    } else {
          id = json['id'];
    }
    if (json['title'] == null) {
      title = null;
    } else {
          title = json['title'];
    }
    if (json['averageRating'] == null) {
      averageRating = null;
    } else {
          averageRating = json['averageRating'];
    }
    if (json['description'] == null) {
      description = null;
    } else {
          description = json['description'];
    }
    if (json['imageUrl'] == null) {
      imageUrl = null;
    } else {
          imageUrl = json['imageUrl'];
    }
    if (json['link'] == null) {
      link = null;
    } else {
          link = json['link'];
    }
    if (json['price'] == null) {
      price = null;
    } else {
          price = json['price'];
    }
    if (json['ratingCount'] == null) {
      ratingCount = null;
    } else {
          ratingCount = json['ratingCount'];
    }
    if (json['score'] == null) {
      score = null;
    } else {
          score = json['score'];
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (id != null)
      json['id'] = id;
    if (title != null)
      json['title'] = title;
    if (averageRating != null)
      json['averageRating'] = averageRating;
    if (description != null)
      json['description'] = description;
    if (imageUrl != null)
      json['imageUrl'] = imageUrl;
    if (link != null)
      json['link'] = link;
    if (price != null)
      json['price'] = price;
    if (ratingCount != null)
      json['ratingCount'] = ratingCount;
    if (score != null)
      json['score'] = score;
    return json;
  }

  static List<InlineResponse20047RecommendedWines> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse20047RecommendedWines>() : json.map((value) => new InlineResponse20047RecommendedWines.fromJson(value)).toList();
  }

  static Map<String, InlineResponse20047RecommendedWines> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse20047RecommendedWines>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse20047RecommendedWines.fromJson(value));
    }
    return map;
  }
}

File: csharp/src/main/CsharpDotNet2/spoonacular/Model/InlineResponse20047RecommendedWines.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace spoonacular.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class InlineResponse20047RecommendedWines {
    /// <summary>
    /// Gets or Sets Id
    /// </summary>
    [DataMember(Name="id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "id")]
    public int? Id { get; set; }

    /// <summary>
    /// Gets or Sets Title
    /// </summary>
    [DataMember(Name="title", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "title")]
    public string Title { get; set; }

    /// <summary>
    /// Gets or Sets AverageRating
    /// </summary>
    [DataMember(Name="averageRating", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "averageRating")]
    public decimal? AverageRating { get; set; }

    /// <summary>
    /// Gets or Sets Description
    /// </summary>
    [DataMember(Name="description", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "description")]
    public string Description { get; set; }

    /// <summary>
    /// Gets or Sets ImageUrl
    /// </summary>
    [DataMember(Name="imageUrl", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "imageUrl")]
    public string ImageUrl { get; set; }

    /// <summary>
    /// Gets or Sets Link
    /// </summary>
    [DataMember(Name="link", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "link")]
    public string Link { get; set; }

    /// <summary>
    /// Gets or Sets Price
    /// </summary>
    [DataMember(Name="price", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "price")]
    public string Price { get; set; }

    /// <summary>
    /// Gets or Sets RatingCount
    /// </summary>
    [DataMember(Name="ratingCount", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ratingCount")]
    public int? RatingCount { get; set; }

    /// <summary>
    /// Gets or Sets Score
    /// </summary>
    [DataMember(Name="score", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "score")]
    public decimal? Score { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class InlineResponse20047RecommendedWines {\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("  Title: ").Append(Title).Append("\n");
      sb.Append("  AverageRating: ").Append(AverageRating).Append("\n");
      sb.Append("  Description: ").Append(Description).Append("\n");
      sb.Append("  ImageUrl: ").Append(ImageUrl).Append("\n");
      sb.Append("  Link: ").Append(Link).Append("\n");
      sb.Append("  Price: ").Append(Price).Append("\n");
      sb.Append("  RatingCount: ").Append(RatingCount).Append("\n");
      sb.Append("  Score: ").Append(Score).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
